# Housing Market

In [10]:
import pandas as pd
import numpy as np
#data1 = np.random.randint(1, high=5, size=100, dtype='l')
#data2 = np.random.randint(1, high=4, size=100, dtype='l')
#data3 = np.random.randint(10000, high=30001, size=100, dtype='l')
#array类型不好处理
s1 = pd.Series(np.random.randint(1, high=5, size=100, dtype='l'))
s2 = pd.Series(np.random.randint(1, high=4, size=100, dtype='l'))
s3 = pd.Series(np.random.randint(10000, high=30001, size=100, dtype='l'))
#series在pandas里更容易处理

### Step 3. Let's create a DataFrame by joinning the Series by column

In [11]:
#series可以直接concat
housemkt = pd.concat([s1, s2, s3], axis=1)
housemkt.head()

0  1      2
0  2  2  14401
1  3  2  22099
2  3  3  10072
3  1  3  26501
4  1  3  13498

### Step 4. Change the name of the columns to bedrs, bathrs, price_sqr_meter

In [12]:
#改名：rename（columns=字典,inplace=True）
housemkt.rename(columns = {0: 'bedrs', 1: 'bathrs', 2: 'price_sqr_meter'}, inplace=True)
housemkt.head()

bedrs  bathrs  price_sqr_meter
0      2       2            14401
1      3       2            22099
2      3       3            10072
3      1       3            26501
4      1       3            13498

### Step 5. Create a one column DataFrame with the values of the 3 Series and assign it to 'bigcolumn'

In [16]:
# join concat the values;axis=0就是竖拼
bigcolumn = pd.concat([s1, s2, s3], axis=0)
# it is still a Series, so we need to transform it to a DataFrame: to_frame()
bigcolumn = bigcolumn.to_frame()

### Step 6. Ops it seems it is going only until index 99. Is it true?

In [18]:
#reset_index
bigcolumn.reset_index(drop=True, inplace=True)
bigcolumn

0
0        2
1        3
2        3
3        1
4        1
5        2
6        3
7        4
8        1
9        4
10       3
11       4
12       3
13       2
14       3
15       4
16       2
17       2
18       2
19       3
20       4
21       2
22       4
23       1
24       4
25       1
26       1
27       4
28       3
29       4
..     ...
270  13538
271  10107
272  10185
273  29610
274  10117
275  27913
276  14068
277  26363
278  19890
279  27197
280  28064
281  28949
282  23652
283  17064
284  29423
285  20660
286  21484
287  24490
288  13055
289  11168
290  23494
291  14272
292  13272
293  16675
294  17123
295  18497
296  14808
297  15463
298  26611
299  24053

[300 rows x 1 columns]

### Step 7. Reindex the DataFrame so it goes from 0 to 299